opennms-correlation/drools-correlation-engine/src/main/java/org/opennms/netmgt/correlation/drools/DroolsCorrelationEngine.java

```diff
@@ -214,11 +224,29 @@ public void initialize() throws Exception {
         if (m_isStreaming) {
             new Thread(() -> {
                 Logging.putPrefix(getClass().getSimpleName() + '-' + getName());
-                m_kieSession.fireUntilHalt();
+                try {
+                    m_kieSession.fireUntilHalt();
+                } catch (Exception e) {
+                    LOG.error("Exception while running rules, reloading engine ", e);
+                    triggerAlarm(e);
+                    reloadConfig();
+                }
             }, "FireTask").start();
         }
     }
 
+    private void triggerAlarm(Exception exception) {
+        EventBuilder ebldr = new EventBuilder(EventConstants.DROOLS_ENGINE_ENCOUNTERED_EXCEPTION, getName());
+        ebldr.addParam(EventConstants.PARM_STACK_TRACE_STRING, ExceptionUtils.getStackTrace(exception));
+        // Add alarmData with alarm type as notification.
```

Comment:
We should probably define the alarm data within the configuration as to hard-coding it here.

For `foundation-2017`, we could add this to a new file such as  `etc/eventconf/opennms.correlator.events.xml`, but omit to include this in the `eventconf.xml` by default since that will surely cause conflicts.

In `release-23.0.1`, we could then update add the import to the `eventconf.xml`.